packages/shared/src/hooks/usePostById.ts

```diff
@@ -30,20 +30,28 @@ export const getPostByIdKey = (id: string): QueryKey => [POST_KEY, id];
 export const updatePostCache = (
   client: QueryClient,
   id: string,
-  { id: _, ...postUpdate }: Partial<Post>,
+  postUpdate:
+    | Partial<Omit<Post, 'id'>>
+    | ((current: Post) => Partial<Omit<Post, 'id'>>),
```

Comment:
why is this needed?

Reply:
`setQueryData` can accept a new value or callback which will get the current value and needs to produce a new one. This change adds support for latter while still keeping compatibility with existing usages.